cmd/piper.go

```diff
@@ -119,6 +127,10 @@ func PrepareConfig(cmd *cobra.Command, metadata *config.StepData, stepName strin
 		}
 	}
 
+	if fmt.Sprintf("%v", stepConfig.Config["collectTelemetryData"]) == "false" {
+		GeneralConfig.NoTelemetry = true
```

Comment:
wording: we have a positive and a negative wording for the telemetry related booleans.
- collectTelemetryData and
- noTelemetry

Looks more confusing than it needs to be  ... Why not having only one of both?

Reply:
We're using a boolean flag here provided by the command framework. So by providing `--noTelemetry` the flag is set to `true`. I guess this could also be implemented as `--collectTelemetryData=false`.

Reply:
I would want to use a boolean flag intuitively, i.e. if set `true`, if not set `false` as we do for `verbose` as well.

Here we do have the documented legacy of the configuration parameter `collectTelemetryData` which we should not break.

Thus suggested behavior is still okay from my POV and good enough to go. In case we should get feedback further down the road we may think of adding noTelemetry also to the configuration which I think we should avoid for now.

Calling the step will provide the proper flag information, checking the docs will provide the proper configuration parameter ...